### PR TITLE
Barkha_Assignment_34

### PR DESCRIPTION
**Pull request by BarkhaPJun20**
Program 1:
image

Program 2:
image

### CODE REVIEW

File: src/barkhaPJun20/ArrayListFunctions_RemoveWordFromArrayList.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	void removeWordfromArrayList(ArrayList <String>arrayList, String removeWord) {
+		while (arrayList.contains(removeWord)) {             			// will iterate loop on "Techno" itself (3 times)
+			arrayList.remove(removeWord);
+		}
+		System.out.println("New ArrayList after remving "+removeWord+" is :"+arrayList);	
+	}
+	//method 2
+	void removeWordfromArrayList1(ArrayList <String>arrayList, String removeWord) {
+		for (int index=0; index<arrayList.size(); index++) {            // will iterate loop till arrayList.size() (6 times)
+			if (arrayList.get(index)==removeWord) {
```
Comment (harshvegada) [inline src/barkhaPJun20/ArrayListFunctions_RemoveWordFromArrayList.java]: Do not use == while comparing String == checks the reference not value, always check with .equals() method

File: src/barkhaPJun20/ArrayListFunctions_ReplaceWordFromArrayList.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.util.ArrayList;
+
+public class ArrayListFunctions_ReplaceWordFromArrayList {
+	int count=0;
+	
+	void replaceWordInArrayList(ArrayList<String> arrayList, String removeWord, String replaceWith) {
+		System.out.print("Index of "+removeWord+" in given ArrayList is:");
+		for (int index=0; index<arrayList.size(); index++) {
+			if (arrayList.get(index)==removeWord) {
```
Comment (harshvegada) [inline src/barkhaPJun20/ArrayListFunctions_ReplaceWordFromArrayList.java]: Do not use == while comparing String == checks the reference not value, always check with .equals() method

File: src/barkhaPJun20/ArrayListFunctions_ReplaceWordFromArrayList.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+				if (arrayList.get(index)==removeWord) {
+					arrayList.set(index, replaceWith);
+				}
+			}	
+		System.out.println(arrayList);	
+					
+		System.out.println("--------------------------------------------------------------------------------------");	
+		
+		for (int index=0; index<arrayList.size(); index++) {
+			if (arrayList.get(index)==replaceWith)
```
Comment (harshvegada) [inline src/barkhaPJun20/ArrayListFunctions_ReplaceWordFromArrayList.java]: Do not use == while comparing String == checks the reference not value, always check with .equals() method

File: src/barkhaPJun20/ArrayListFunctions_RemoveWordFromArrayList.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+["Maulik","Techno" ,"Krishna", "Techno", "Credits", "Techno"]
+Hint : contains & remove*/
+
+package barkhaPJun20;
+
+import java.util.ArrayList;
+
+public class ArrayListFunctions_RemoveWordFromArrayList {
+	//method 1
+	void removeWordfromArrayList(ArrayList <String>arrayList, String removeWord) {
```
Comment (harshvegada) [inline src/barkhaPJun20/ArrayListFunctions_RemoveWordFromArrayList.java:12]: You can manage whole logic in single for loop, like below

you have iterate on for loop and check in if condition 

if (names.get(index).equals(indexName)) {
	names.set(names.indexOf("Techno"), "Credits");
	countReplacement++;
} else if (names.get(index).equals(replacement)) {
        countReplacement++;
}

**Review by harshvegada — COMMENTED** [review]
Incorporate changes

File: src/barkhaPJun20/ArrayListFunctions_RemoveWordFromArrayList.java (hunk 4 above)
Comment (BarkhaPJun20) [reply src/barkhaPJun20/ArrayListFunctions_RemoveWordFromArrayList.java:12]: Incorporated changes as suggested

File: src/barkhaPJun20/ArrayListFunctions_RemoveWordFromArrayList.java (hunk 1 above)
Comment (BarkhaPJun20) [reply src/barkhaPJun20/ArrayListFunctions_RemoveWordFromArrayList.java]: Yes...agreed... changesDone

File: src/barkhaPJun20/ArrayListFunctions_ReplaceWordFromArrayList.java (hunk 2 above)
Comment (BarkhaPJun20) [reply src/barkhaPJun20/ArrayListFunctions_ReplaceWordFromArrayList.java]: es...agreed... changedDone

**Review by harshvegada — APPROVED** [review]
Code looks Good @BarkhaPJun20 
Keep it up

**Review by KrishnaKTechnocredits — APPROVED** [review]
Good job dear.

**Pull request merged**